packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js

```diff
@@ -59,107 +62,289 @@ describe('pocl-validation-errors', () => {
   })
 
   describe('when no validation errors are returned from Sales Api', () => {
-    let result
-    beforeEach(async () => {
+    beforeAll(() => {
       salesApi.getPoclValidationErrorsForProcessing.mockResolvedValue(null)
-      result = await processPoclValidationErrors()
     })
 
-    it('returns undefined', () => {
-      expect(result).toBe(undefined)
+    it('returns undefined', async () => {
+      const result = await processPoclValidationErrors()
+      expect(result).toBeUndefined()
     })
 
-    it('does not attempt to create transactions', () => {
+    it('does not attempt to create transactions', async () => {
+      await processPoclValidationErrors()
       expect(salesApi.createTransactions).not.toBeCalled()
     })
 
-    it('does not finalise transactions which have failed', () => {
+    it('does not finalise transactions which have failed', async () => {
+      await processPoclValidationErrors()
       expect(salesApi.finaliseTransaction).not.toBeCalled()
     })
   })
 
   describe('processes successfully fixed validation errors', () => {
-    let poclValidationError
-    beforeEach(async () => {
-      poclValidationError = getPoclValidationError()
-      salesApi.getPoclValidationErrorsForProcessing.mockResolvedValue([poclValidationError])
-      salesApi.createTransactions.mockResolvedValue([{ statusCode: 201, response: { id: 'test-response-id' } }])
+    beforeAll(() => {
+      salesApi.getPoclValidationErrorsForProcessing.mockResolvedValue([getPoclValidationError()])
```

Comment:
This needs to change to `mockImplementation` so a fresh value is available every time